File: textbooks/SICP_JS/chapter_1/exercise_1_4.js
// exrcise 1.4
function plus(a, b) { return a + b; }

function minus(a, b) { return a - b; }

function a_plus_abs_b(a, b) {
    return (b >= 0 ? plus : minus)(a, b);
}

console.log(a_plus_abs_b(2, -2))  // 4
console.log(a_plus_abs_b(2, 3))  // 5
console.log(a_plus_abs_b(2, 0))  // 2

File: textbooks/SICP_go_edition/section_1/exercise_1_4.go
package main

import "fmt"

func Plus(a, b int) int {
	return a + b
}

func Minus(a, b int) int {
	return a - b
}

func APlusAbsB(a, b int) int {
	if b >= 0 {
		return Plus(a, b)
	} else {
		return Minus(a, b)
	}
}

func main() {
	fmt.Println(APlusAbsB(2, -2))
	fmt.Println(APlusAbsB(2, 3))
	fmt.Println(APlusAbsB(2, 0))
}
